packages/ketchup/src/components/kup-activity-timeline/kup-activity-timeline.tsx

```diff
@@ -0,0 +1,393 @@
+import {
+    Component,
+    Element,
+    Event,
+    EventEmitter,
+    forceUpdate,
+    Fragment,
+    h,
+    Host,
+    Method,
+    Prop,
+    State,
+    VNode,
+    Watch,
+} from '@stencil/core';
+import {
+    KupManager,
+    kupManagerInstance,
+} from '../../managers/kup-manager/kup-manager';
+import { KupDataDataset } from '../../managers/kup-data/kup-data-declarations';
+import { componentWrapperId } from '../../variables/GenericVariables';
+import {
+    GenericObject,
+    KupComponent,
+    KupEventPayload,
+} from '../../types/GenericTypes';
+import { getProps, setProps } from '../../utils/utils';
+import {
+    ActivityTimeline,
+    KupActivity,
+    KupActivityTimelineAction,
+    KupActivityTimelineData,
+    KupActivityTimelineProps,
+    KupDatatableClickEventPayload,
+} from './kup-activity-timeline-declarations';
+import { FImage } from '../../f-components/f-image/f-image';
+import { getCellValueForDisplay } from '../../utils/cell-utils';
+
+@Component({
+    tag: 'kup-activity-timeline',
+    styleUrl: 'kup-activity-timeline.scss',
+    shadow: true,
+})
+export class KupActivityTimeline {
+    /**
+     * References the root HTML element of the component (<kup-activity-timeline>).
+     */
+    @Element() rootElement: HTMLElement;
+
+    /*-------------------------------------------------*/
+    /*                   S t a t e s                   */
+    /*-------------------------------------------------*/
+
+    @State()
+    timeline: ActivityTimeline[] = [];
+
+    /*-------------------------------------------------*/
+    /*                    P r o p s                    */
+    /*-------------------------------------------------*/
+
+    /**
+     *  Dataset containing the activities list.
+     * @default null
+     */
+    @Prop() data: KupDataDataset;
+
+    /**
+     *  Columns containing dates.
+     * @default null
+     */
+    @Prop() dateColumn: string;
+
+    /**
+     *  Columns containing times.
+     * @default null
+     */
+    @Prop() timeColumn: string;
+    /*-------------------------------------------------*/
+    /*       I n t e r n a l   V a r i a b l e s       */
+    /*-------------------------------------------------*/
+
+    #kupManager: KupManager = kupManagerInstance();
+
+    /*-------------------------------------------------*/
+    /*                   E v e n t s                   */
+    /*-------------------------------------------------*/
+
+    /**
+     * Generic click event on activity.
+     */
+    @Event({
```

Comment:
The event names are wrong, the naming convention is `kul-componentname-eventname`.